### PR TITLE
Improving tr-suite-master

### PR DESCRIPTION
**Pull request by emesika**
This patch refactors and improves the tr-suite-master suite by:

1. Using the ansible-suite-master code when available, this has two exceptions :
   a) ost.json is copied since we need more memory and vCPUs for the hosts.
   b) test_002_ansible.py is copied to shorten suite runtime bt cutting redundant stages

2. The ansible-suite-master is no longer compiling the OCP installer binary, this is now done
   by a workflow in terraform-provider-repo repository under the v0 branch. The workflow compiles
   both OCP installer and Terraform Provide PlugIn with the patch changes and publish them as a
   tr-ocp-installer rpm that is consumed by the oVirt engine VM created by OST

### CODE REVIEW

File: tr-suite-master/test-run-ocp-installer.sh [hunk 1]
```diff
@@ -113,38 +115,37 @@ sed -i -e 's;OVIRT_API_URL;'"$OVIRT_API_URL"';' \
...
 
-cp -f ${WORKING_DIR}/install-config.yaml.in ${ARTIFACTS_DIR}/install-config.yaml
+cp -f ${WORKING_DIR}/install-config.yaml.in ${CONFIG_DIR}/install-config.yaml
 
 # set SSH key
 info "set SSH key, prefix, oVirt cluster id, oVirt SD id , oVirt network name"
 
 if [ ! -e ${SSH_KEY_FILE} ]; then
-    git_cleanup_and_error "ssh key file ${SSH_KEY_FILE} does not esists"
+    error "ssh key file ${SSH_KEY_FILE} does not esists"
```
Comment (tinez) [inline tr-suite-master/test-run-ocp-installer.sh:141]: `esists` -> `exists`

**Review by tinez — COMMENTED** [review]
Some nits

File: tr-suite-master/test-init-terraform.sh [hunk 2]
```diff
@@ -2,147 +2,33 @@
...
 }
 
 #get parameters
-while getopts hw:t: option; do
+while getopts hl: option; do
     case $option in
             h) usage; exit 0;;
-            w) WORKING_DIR="${OPTARG}";;
-            t) TPR_OR_CH_OR_MB="${OPTARG}";;
+            l) PLUGING_DIR="${OPTARG}";;
```
[the remark this answers is not in the document]
Comment (emesika) [reply tr-suite-master/test-init-terraform.sh]: Done

File: tr-suite-master/test-init-terraform.sh [hunk 3]
```diff
@@ -2,147 +2,33 @@
...
     -force-copy \
     -get=true \
     -reconfigure \
     -upgrade=false \
-    -plugin-dir="${GOPKG}"
-
-if [[ $? -ne 0 ]]; then
-    git_cleanup_and_error "failed to init terraform provider on top of  PR ${TPR_OR_CH_OR_MB}, please check log file/s"
-fi
+    -plugin-dir="${PLUGING_DIR}"
```
[the remark this answers is not in the document]
Comment (emesika) [reply tr-suite-master/test-init-terraform.sh]: Done

File: tr-suite-master/test-run-ocp-installer.sh (hunk 1 above)
Comment (emesika) [reply tr-suite-master/test-run-ocp-installer.sh]: Done

File: tr-suite-master/test-scenarios/test_003_run_ocp_installer.py [hunk 4]
```diff
@@ -17,35 +17,28 @@
...
-    pr = os.environ.get('STD_CI_REFSPEC')
-    if pr == None or not pr:
-        pr = "master"
-
+    sds = sds_service.list(search=f'name={sd_name}')
+    sd_id = next(iter([sd.id for sd in sds if sd.name == sd_name]))
+    dc_id = engine.data_centers_service().list(search=f'name={dc_name}')[0].id
+    networks = engine.networks_service().list(search=f'name={network_name}')
+    network_id = next(iter([network.id for network in networks if network.data_center.id == dc_id]))
+    ovirt_vnic_profile_id = engine.networks_service().network_service(network_id).vnic_profiles_service().list()[0].id
```
Comment (erav) [inline tr-suite-master/test-scenarios/test_003_run_ocp_installer.py:41]: same as above

File: tr-suite-master/test-scenarios/test_003_run_ocp_installer.py [hunk 5]
```diff
@@ -17,35 +17,28 @@
...
-
-
-    # get PR number from the env if not found then use master branch
-    pr = os.environ.get('STD_CI_REFSPEC')
-    if pr == None or not pr:
-        pr = "master"
-
+    sds = sds_service.list(search=f'name={sd_name}')
+    sd_id = next(iter([sd.id for sd in sds if sd.name == sd_name]))
+    dc_id = engine.data_centers_service().list(search=f'name={dc_name}')[0].id
```
[the remark this answers is not in the document]
Comment (emesika) [reply tr-suite-master/test-scenarios/test_003_run_ocp_installer.py:38]: Done

File: tr-suite-master/test-scenarios/test_003_run_ocp_installer.py (hunk 4 above)
Comment (emesika) [reply tr-suite-master/test-scenarios/test_003_run_ocp_installer.py:41]: Done

**Review by tinez — APPROVED** [review]

**Review by erav — APPROVED** [review]

**Pull request merged**